### PR TITLE
Add error handling to Scheduler

### LINKED ISSUE #155: Error handling for scheduler
We need to be able to parse the scheduling errors from the server and present them in a user friendly fashion.
Here is the beginning of the work for parsing them: https://github.com/conundrumgirl/mtb/blob/f588910ae46c7e6dc602821a76ffa0faffeb433c/src/components/studies/StudyBuilder.tsx#L216
We should show the errors on the level applicable to it. e.g. if it relates to the time window - highlight the window. If it applies to the session -- outline the session
image

### PR DESCRIPTION
**Pull request by leonli33**
This is how the errors look:
Screen Shot 2021-05-25 at 4 51 56 PM

Resolves #155

### CODE REVIEW

File: src/components/studies/scheduler/AssessmentWindow.tsx [hunk 1]
```diff
@@ -38,39 +38,37 @@ const useStyles = makeStyles(theme => ({
...
   onDelete,
   index,
+  errorText,
 }: AssessmentWindowProps) => {
-
   const classes = useStyles()
   return (
     <Paper
-     className={classes.root}
+      className={clsx(classes.root, errorText && classes.redBorder)}
```
Comment (leonli33) [inline src/components/studies/scheduler/AssessmentWindow.tsx:67]: This is the only thing I added to the component.

File: src/components/studies/scheduler/SchedulableSingleSessionContainer.tsx [hunk 2]
```diff
@@ -45,259 +43,325 @@ type SchedulableSingleSessionContainerProps = {
...
-                {schedulableSession.timeWindows?.map((window, index) => (
-                  <AssessmentWindow
-                    index={index}
-                    key={`${index}${window.startTime}${window.expiration}`}
-                    onDelete={() => {
-                      console.log('deleting1', index)
-                      deleteWindow(index)
+  return (
+    <Box bgcolor="#F8F8F8" flexGrow="1" pb={2.5} pl={4}>
+      {sessionErrorState && sessionErrorState.generalErrorMessage.length > 0 && (
```
Comment (leonli33) [inline src/components/studies/scheduler/SchedulableSingleSessionContainer.tsx:197]: Added error display here

File: src/components/studies/scheduler/SchedulableSingleSessionContainer.tsx [hunk 3]
```diff
@@ -45,259 +43,325 @@ type SchedulableSingleSessionContainerProps = {
...
+                ...schedulableSession,
+                interval,
+              })
+            }}
+            interval={schedulableSession.interval}
+          ></RepeatFrequency>
+        </Box>
+
+        <Box className={classes.formSection}>
+          {windowErrors.length > 0 && (
```
Comment (leonli33) [inline src/components/studies/scheduler/SchedulableSingleSessionContainer.tsx:243]: Added error display here

File: src/components/studies/StudyBuilder.tsx [hunk 4]
```diff
@@ -94,19 +94,18 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
...
-      id: string
-      section: StudySection
-    }>()
+  let { id, section: _section } = useParams<{
+    id: string
+    section: StudySection
+  }>()
   const [section, setSection] = React.useState(_section)
   const [error, setError] = React.useState<string[]>([])
+  const [schedulerError, setSchedulerError] = React.useState<string[]>([])
```
Comment (conundrumgirl) [inline src/components/studies/StudyBuilder.tsx:103]: do you need a separate schedulerError? Can you just use error? When the scheduler is displayed all the errors will be the scheduler error. You can change the code around the ln 383 and check the section and don't display the errors the the studybuilder if the the section is scheduler

File: src/components/studies/StudyBuilder.tsx [hunk 5]
```diff
@@ -211,11 +211,11 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
       setData({
         ...builderInfo,
         schedule: savedUpdatedSchedule,
-        study: {...builderInfo.study, version: builderInfo.study.version+1} //temporary to get notifications working
+        study: { ...builderInfo.study, version: builderInfo.study.version + 1 }, //temporary to get notifications working
       })
       setHasObjectChanged(false)
     } catch (e) {
-      console.log(e)
+      console.log(e, 'error')
```
Comment (conundrumgirl) [inline src/components/studies/StudyBuilder.tsx:218]: I think this logic should happen in the scheduler component. Just set the error and move this logic in the scheduler. 
The reasons for that are that:
a. this logic scheduler specific
b. you end up parsing the errors twice: - first time here and the second time in the scheduler

**Review by conundrumgirl — CHANGES_REQUESTED** [review]

**Review by conundrumgirl — COMMENTED** [review]
let's merge it and see what breaks:)

**Pull request merged**